From 34b541cd6b210da36461a6c8330c92f076496f2f Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Mon, 2 Feb 2009 13:30:01 +0800
Subject: [PATCH] ENGR00106778 Reset the max clock on esdhc port 2

reset the max clock on esdhc port 2, on all the esdhc
mxc platforms.

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 arch/arm/mach-mx35/mx35_3stack.c |    6 +++---
 arch/arm/mach-mx37/mx37_3stack.c |    6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-mx35/mx35_3stack.c b/arch/arm/mach-mx35/mx35_3stack.c
index 45a4f24..e542ed9 100644
--- a/arch/arm/mach-mx35/mx35_3stack.c
+++ b/arch/arm/mach-mx35/mx35_3stack.c
@@ -521,7 +521,7 @@ static struct mxc_mmc_platform_data mmc0_data = {
 #endif
 	.min_clk = 400000,
 	.max_clk = 52000000,
-	.card_inserted_state = 1,
+	.card_inserted_state = 0,
 	.status = sdhc_get_card_det_status,
 	.wp_status = sdhc_write_protect,
 	.clock_mmc = "sdhc_clk",
@@ -566,8 +566,8 @@ static struct mxc_mmc_platform_data mmc1_data = {
 	    MMC_VDD_31_32,
 	.caps = MMC_CAP_4_BIT_DATA,
 	.min_clk = 150000,
-	.max_clk = 25000000,
-	.card_inserted_state = 1,
+	.max_clk = 50000000,
+	.card_inserted_state = 0,
 	.status = sdhc_get_card_det_status,
 	.wp_status = sdhc_write_protect,
 	.clock_mmc = "sdhc_clk",
diff --git a/arch/arm/mach-mx37/mx37_3stack.c b/arch/arm/mach-mx37/mx37_3stack.c
index 80587d5..9edeaad 100644
--- a/arch/arm/mach-mx37/mx37_3stack.c
+++ b/arch/arm/mach-mx37/mx37_3stack.c
@@ -593,7 +593,7 @@ static struct mxc_mmc_platform_data mmc_data = {
 	.caps = MMC_CAP_4_BIT_DATA,
 	.min_clk = 400000,
 	.max_clk = 52000000,
-	.card_inserted_state = 1,
+	.card_inserted_state = 0,
 	.status = sdhc_get_card_det_status,
 	.wp_status = sdhc_write_protect,
 	.clock_mmc = "esdhc_clk",
@@ -638,8 +638,8 @@ static struct mxc_mmc_platform_data mmc1_data = {
 	    MMC_VDD_31_32,
 	.caps = MMC_CAP_4_BIT_DATA,
 	.min_clk = 150000,
-	.max_clk = 25000000,
-	.card_inserted_state = 1,
+	.max_clk = 50000000,
+	.card_inserted_state = 0,
 	.status = sdhc_get_card_det_status,
 	.wp_status = sdhc_write_protect,
 	.clock_mmc = "esdhc_clk",
-- 
1.5.4.4

